File: segundo/1er cuatri/PDOO/irrgarten_entrega1/Irrgarten_java/src/irrgarten/Weapon.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package irrgarten;

/**
 *
 * @author alvarogz
 */
public class Weapon {
    private float power;
    private int uses;
    
    public Weapon(float power, int uses) {
        this.power = power;
        this.uses = uses;
    }
    
    public float attack() {
        if (this.uses > 0){
            this.uses--;
            return this.power;
        }
        else
            return 0;
    }
    
    @Override
    public String toString() {
        return "W[" + this.power + ", " + this.uses + "]";
    }
    
    public boolean discard() {
        return Dice.discardElement(uses);
    }
}

File: segundo/1er cuatri/PDOO/irrgarten_entrega1/irrgarten_ruby/Irrgarten/weapon.rb
# frozen_string_literal: true
require_relative 'dice'

class Weapon
  def initialize(power, uses)
    @power = power
    @uses = uses

  end

  def attack
    if @uses > 0
      @uses = @uses-1
      @power
    else
      0.to_f
    end
  end

  def to_s
    "W[#{@power}, #{@uses}]"
  end

  def inspect
    to_s
  end

  def discard
    Dice.discard_element(@uses)
  end
end
